refact: enhance state management and full typescript support

aminbenmansour opened the pull request:
cleaner and more readable code

aminbenmansour committed:
using prettier for uniform indentation

aminbenmansour committed:
adding redux for centralized state management

aminbenmansour committed:
refact: full implementation with redux

aminbenmansour committed:
refact: using context api for managing AID

aminbenmansour committed:
refact: using context api for managing ACDC

aminbenmansour committed:
refact: isolating separate component

aminbenmansour committed:
refact: full typescript support, comment un-used code, naming conventions

aminbenmansour committed:
refact: further decoupling components into files

AlexAndrei98 reviewed (commented):
Thank you for cleaning up the code and actually make it fully typescript! ! It is much better organized!

AlexAndrei98 commented on src/types/aid.ts:
```diff
@@ -0,0 +1,17 @@
+  salty: Salty;
+}
+
+interface Salty {
```
I think these types can also be added to the native signify-ts client rather than in the web app itself! It would be awesome if you could make a pr [here](https://github.com/WebOfTrust/signify-ts/blob/main/src/keri/core/manager.ts#L112) as well to add these!

aminbenmansour replied:
Sure! Thank you for pointing out to this!

aminbenmansour commented:
I thought it would be more understandable for newcomers and much easier and less intimidating to add new features if the code was strongly typed and split into a hierarchy :)

2byrds commented:
I plan to review/test this tonight or tomorrow

aminbenmansour committed:
refact: managing the modal state and error state with redux

aminbenmansour committed:
refact: renaming shared-slice to auth-slice

All variables in shared-slice are related to authentication, thus, auth-slice is more semantic. We can have a shared-slice later if needed

2byrds commented:
I had some issues running the orig repo (and yours) but they are on my side. I'll come back to this soon.

aminbenmansour commented:
please reinstall dependencies using `yarn`

AlexAndrei98 commented:
@2byrds have you had a chance to integration test this PR?

2byrds commented:
No, I've been buried in other tasks. I'll try to revisit soon, thank you for the reminder!
